network/android/src/main/java/com/capacitorjs/plugins/network/Network.java

```diff
@@ -59,25 +86,66 @@ public NetworkStatusChangeListener getStatusChangeListener() {
 
     /**
      * Get the current network information.
-     * @return NetworkInfo
+     * @return NetworkStatus
      */
-    public NetworkInfo getNetworkStatus() {
-        return connectivityManager.getActiveNetworkInfo();
+    @SuppressWarnings("deprecation")
+    public NetworkStatus getNetworkStatus() {
+        NetworkStatus networkStatus = new NetworkStatus();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            if (this.connectivityManager != null) {
+                NetworkCapabilities capabilities =
+                    this.connectivityManager.getNetworkCapabilities(this.connectivityManager.getActiveNetwork());
+                if (this.connectivityManager.getActiveNetwork() != null && capabilities != null) {
+                    networkStatus.connected = capabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET);
+                    if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
+                        networkStatus.connectionType = NetworkStatus.ConnectionType.WIFI;
+                    } else if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
+                        networkStatus.connectionType = NetworkStatus.ConnectionType.CELLULAR;
+                    } else {
+                        networkStatus.connectionType = NetworkStatus.ConnectionType.UNKNOWN;
+                    }
+                }
+            }
+        } else {
+            NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+            if (networkInfo != null) {
+                networkStatus.connected = networkInfo.isConnected();
+                String typeName = networkInfo.getTypeName();
+                if (typeName.equals("WIFI")) {
+                    networkStatus.connectionType = NetworkStatus.ConnectionType.WIFI;
+                } else if (typeName.equals("MOBILE")) {
+                    networkStatus.connectionType = NetworkStatus.ConnectionType.CELLULAR;
+                }
+            }
+        }
+        return networkStatus;
     }
 
     /**
-     * Register a Intent Receiver in the activity.
-     * @param activity
+     * Register a network callback.
      */
-    public void startMonitoring(@NonNull AppCompatActivity activity) {
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    public void startMonitoring() {
+        connectivityManager.registerDefaultNetworkCallback(connectivityCallback);
+    }
+
+    @SuppressWarnings("deprecation")
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP_MR1)
+    public void startMonitoring(AppCompatActivity activity) {
         IntentFilter filter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
```

Comment:
instead of suppressing the warning you can use the actual constant value `"android.net.conn.CONNECTIVITY_CHANGE"`